# U.S Medical Insurance Costs

## **Overview**

This U.S cost of medical insurance can be dependes on a person's age, sex, body mass index (IBM), the number of children they have, where they are from and whether or not they smoke

## **Scope**

Here, I will be using Python3 fundamentals to analyze the dataset containing medical insurance data to determine how various factors can impact the cost of medical insurance. And test my pervious python learning course until now    

In [1]:
# importing cvs file
import csv 

here this importing csv file is important to read insurance.csv dataset 

In [2]:
with open('/content/drive/MyDrive/python-portfolio-example-solution/insurance.csv') as insurance_file:
  insurance_dataset = insurance_file.read()

This insurance.csv file contain 7 columns:

*   Patient's age 
*   Patient's sex
*   Patient's BMI
*   Patient's number of children
*   Patient's smoking status
*   Patient's location
*   Patient's medical insurance cost

due to the file is stored in csv file it is difficult to manipulate and analyze. so we need to create and store into list file for each columns.



In [3]:
# Create empty lists for each column of data in csv file
ages = []
sexes = []
bmis = []
num_children = []
smoker_status = []
regions = []
charges = []

In [4]:
# sort data into lists
with open('/content/drive/MyDrive/python-portfolio-example-solution/insurance.csv', newline = '') as insurance_file:
    insurance_dict = csv.DictReader(insurance_file)
    for row in insurance_dict:
        ages.append(row['age'])
        sexes.append(row['sex'])
        bmis.append(row['bmi'])
        num_children.append(row['children'])
        smoker_status.append(row['smoker'])
        regions.append(row['region'])
        charges.append(row['charges'])

see how the data is puted in our list file

In [8]:
print('Age' + str(ages[:5]))
print('Sex' + str(sexes[:5]))
print('BMI' + str(bmis[:5]))
print('Number of children' + str(num_children[:5]))
print('Smoker Status' + str(smoker_status[:5]))
print('Regions' + str(regions[:5]))
print('Charges' + str(charges[:5]))

Age['19', '18', '28', '33', '32']
Sex['female', 'male', 'male', 'male', 'male']
BMI['27.9', '33.77', '33', '22.705', '28.88']
Number of children['0', '1', '3', '0', '0']
Smoker Status['yes', 'no', 'no', 'no', 'no']
Regions['southwest', 'southeast', 'southeast', 'northwest', 'northwest']
Charges['16884.924', '1725.5523', '4449.462', '21984.47061', '3866.8552']


Check if there is no any missing data is there inthe list 

In [10]:
#verify that each column does not have nulls and is not missing any information
print(len(ages) == (len(sexes)) == (len(bmis)) == (len(num_children)) == (len(smoker_status)) == (len(regions)) == (len(charges)))

True


As we see all length of the data is the same so we can conclude that there is no any missing data in there

In [11]:
print(len(ages))

1338


To understand our dataset more we will do avergae from the given data

In [12]:
def Calculate_average(datas):
    total = 0
    for x in datas:
        total += float(x)
    return float(total/(len(datas)))

print(Calculate_average(ages))
print(Calculate_average(bmis))
print(Calculate_average(num_children))
print(Calculate_average(charges))

39.20702541106129
30.663396860986538
1.0949177877429
13270.422265141257


Avergae datas shows that $13270.42 dollar has charged for average age is 39,
average bmi 30, and only one chlidren.

next to this we will add sex factor into the dataset

In [13]:
male_count = 0
female_count = 0
for sex in sexes:
    if sex == "male":
        male_count += 1
    else:
        female_count += 1

print("There are {male_count} males in the dataset.".format(male_count = male_count))
print("There are {female_count} females in the dataset.".format(female_count = female_count))

There are 676 males in the dataset.
There are 662 females in the dataset.


from 1338 register Patient's 676 are male and the others are Female.

In [14]:
region_count = {}
for region in regions:
    if region not in region_count:
        region_count[region] = 1
    else:
        region_count[region] += 1

print(region_count)

{'southwest': 325, 'southeast': 364, 'northwest': 325, 'northeast': 324}


In [15]:
smoker_count = 0
nonsmoker_count = 0
for data in smoker_status:
    if data == "yes":
        smoker_count += 1
    else:
        nonsmoker_count += 1

print("There are {smoker_count} smokers in the dataset.".format(smoker_count = smoker_count))
print("There are {nonsmoker_count} nonsmokers in the dataset.".format(nonsmoker_count = nonsmoker_count))

There are 274 smokers in the dataset.
There are 1064 nonsmokers in the dataset.


From 1338 Patient's 274 are smokers and the other's are not 

In [18]:
# now let's compare our data based on sex, male and female 

def compare_costs(cost_factor, factor1, factor2):
    with open("/content/drive/MyDrive/python-portfolio-example-solution/insurance.csv") as f:
        file_dict = csv.DictReader(f)
        total_factor1_charges = 0
        factor1_count = 0
        total_factor2_charges = 0
        factor2_count = 0
        for row in file_dict:
            if row[cost_factor] == factor1:
                total_factor1_charges += float(row['charges'])
                factor1_count += 1
            elif row[cost_factor] == factor2:
                total_factor2_charges += float(row['charges'])
                factor2_count += 1
                
    average_factor1_cost = round((total_factor1_charges / factor1_count), 2)
    average_factor2_cost = round((total_factor2_charges / factor2_count), 2)
             
    print("Average insurance costs if {cost_factor} is {factor1}: ${average_factor1_cost}"
          .format(cost_factor = cost_factor, factor1 = factor1, average_factor1_cost = (average_factor1_cost)))
    print("Average insurance costs if {cost_factor} is {factor2}: ${average_factor2_cost}"
          .format(cost_factor = cost_factor, factor2 = factor2, average_factor2_cost = average_factor2_cost))
    
    cost_difference = round(average_factor1_cost - average_factor2_cost, 2)
                    
    print ("The difference in costs depending on {cost_factor} is ${cost_difference}"
           .format(cost_factor = cost_factor, cost_difference = cost_difference))

In [19]:
insurance_diff_for_sex = compare_costs('sex', 'male', 'female')

Average insurance costs if sex is male: $13956.75
Average insurance costs if sex is female: $12569.58
The difference in costs depending on sex is $1387.17


Depend on sex the average cost is $1387.17
 

*   Female - avergae costs $12569.58

*   Male - avergae costs- $13926.75

as we see the male avergae cost is gerder than the femal cost



In [20]:
insurance_diff_for_smoker_status = compare_costs('smoker', 'yes', 'no')

Average insurance costs if smoker is yes: $32050.23
Average insurance costs if smoker is no: $8434.27
The difference in costs depending on smoker is $23615.96


Here the average cost with the effect of smoker is $23615.96 

*   Smoker - 32050.23$ average cost

*   Not Smoker - 8434.27$ average cost



In [21]:
with open('/content/drive/MyDrive/python-portfolio-example-solution/insurance.csv') as insurance_file:
    insurance_dict = csv.DictReader(insurance_file)
    underweight_count = 0
    healthy_count = 0
    overweight_count = 0
    obese_count = 0
    
    underweight_charges = 0
    healthy_charges = 0
    overweight_charges = 0
    obese_charges = 0
    
    for row in insurance_dict:
        bmi = float(row['bmi'])
        charge = float(row['charges'])
        if bmi < 18.5:
            underweight_count += 1
            underweight_charges += charge
        elif bmi >= 18.5 and bmi < 25:
            healthy_count += 1
            healthy_charges += charge
        elif bmi >= 25 and bmi < 30:
            overweight_count += 1
            overweight_charges += charge
        else:
            obese_count += 1
            obese_charges += charge
    
    avg_underweight_charges = round(underweight_charges/ underweight_count, 2)
    avg_healthy_charges = round(healthy_charges / healthy_count, 2)
    avg_overweight_charges = round(overweight_charges / overweight_count, 2)
    avg_obese_charges = round(obese_charges / obese_count)
    
print("The average cost for patients in the underweight range is : $" + str(avg_underweight_charges))
print("The average cost for patients in the healthy range is : $" + str(avg_healthy_charges))
print("The average cost for patients in the overweight range is : $" + str(avg_overweight_charges))
print("The average cost for patients in the obese range is : $" + str(avg_obese_charges))

print("The difference in average cost for obese patients versus overweight patients is: $" + str(avg_obese_charges - avg_overweight_charges))

The average cost for patients in the underweight range is : $8852.2
The average cost for patients in the healthy range is : $10409.34
The average cost for patients in the overweight range is : $10987.51
The average cost for patients in the obese range is : $15552
The difference in average cost for obese patients versus overweight patients is: $4564.49


From our analysis above, we can see that patients in the obese range pay $4565.49 than patients in the overweight range.